### PR TITLE
Stabilized in small data

### PR DESCRIPTION
**Pull request by JeremyLi28**
Seems stabilized in my machine.
Needs further test and test on big data.
Needs further styling

### CODE REVIEW

File: src/main/resources/twitter-demo/newTweetBook.tpl [hunk 1]
```diff
@@ -0,0 +1,139 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset='utf-8' />
+    <title></title>
```
Comment (JavierJia) [inline src/main/resources/twitter-demo/newTweetBook.tpl]: remove it or give it a title?

File: src/main/resources/twitter-demo/newTweetBook.tpl [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+    <script src="static/js/asterix-sdk-stable.js"></script>
+    <script src="static/js/rainbowvis.js"></script>
+    <script src="http://code.jquery.com/jquery.min.js"></script>
+     <script src="static/js/bootstrap.min.js"></script>
+     <script src='https://api.tiles.mapbox.com/mapbox-gl-js/v0.14.0/mapbox-gl.js'></script>
+     <script src='https://api.mapbox.com/mapbox.js/v2.3.0/mapbox.js'></script>
+     <script src="http://cdn.leafletjs.com/leaflet/v0.7.7/leaflet.js"></script>
+     <script src="//d3js.org/d3.v3.min.js" charset="utf-8"></script>
+      <script src="static/js/crossfilter.min.js"></script>
```
Comment (JavierJia) [inline src/main/resources/twitter-demo/newTweetBook.tpl]: use some code formatting tools to unify the indentation?

File: src/main/resources/twitter-demo/newTweetBook.tpl [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    <link href='https://api.tiles.mapbox.com/mapbox-gl-js/v0.14.0/mapbox-gl.css' rel='stylesheet' />
+    <link href="//netdna.bootstrapcdn.com/bootstrap/3.0.0/css/bootstrap.min.css" rel="stylesheet" type="text/css">
+     <link href="static/css/dc.css" rel="stylesheet" type="text/css">
+     <link rel="stylesheet" href="http://cdn.leafletjs.com/leaflet/v0.7.7/leaflet.css" />
+     <link href='https://api.mapbox.com/mapbox.js/v2.3.0/mapbox.css' rel='stylesheet' />
+
+    <script src="static/js/asterix-sdk-stable.js"></script>
+    <script src="static/js/rainbowvis.js"></script>
+    <script src="http://code.jquery.com/jquery.min.js"></script>
+     <script src="static/js/bootstrap.min.js"></script>
```
Comment (JavierJia) [inline src/main/resources/twitter-demo/newTweetBook.tpl]: there is one at line 9, is this one still necessary?

File: src/main/resources/twitter-demo/newTweetBook.tpl (hunk 2 above)
Comment (JeremyLi28) [inline src/main/resources/twitter-demo/newTweetBook.tpl]: Sorry, it seems that my sublime's indentation is too small, can't tell difference

File: src/main/resources/twitter-demo/newTweetBook.tpl [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+     <script src='https://api.tiles.mapbox.com/mapbox-gl-js/v0.14.0/mapbox-gl.js'></script>
+     <script src='https://api.mapbox.com/mapbox.js/v2.3.0/mapbox.js'></script>
```
Comment (JavierJia) [inline src/main/resources/twitter-demo/newTweetBook.tpl]: what's the differences between libs in line 18 and 19?

File: src/main/resources/twitter-demo/run_tweetbook_demo.py [hunk 5]
```diff
@@ -30,7 +30,8 @@
 # Core Routing
 @route('/')
 def jsontest():
-    return template('tweetbook')
+    # return template('tweetbook')
```
Comment (JavierJia) [inline src/main/resources/twitter-demo/run_tweetbook_demo.py]: you can direct overwrite the original `tweetbook`

File: src/main/resources/twitter-demo/static/js/newTweetBook.js [hunk 6]
```diff
@@ -0,0 +1,655 @@
...
+    }
+  });
+
+  // UI-Elements: create map and add controls
+  map = L.map('map').setView([39.5,-96.35,], 4);
+  L.tileLayer('https://api.tiles.mapbox.com/v4/{id}/{z}/{x}/{y}.png?access_token={accessToken}', {
+      attribution: 'Map data &copy; <a href="http://openstreetmap.org">OpenStreetMap</a> contributors, <a href="http://creativecommons.org/licenses/by-sa/2.0/">CC-BY-SA</a>, Imagery © <a href="http://mapbox.com">Mapbox</a>',
+      maxZoom: 18,
+      id: 'jeremyli.p6f712pj',
+      accessToken: 'pk.eyJ1IjoiamVyZW15bGkiLCJhIjoiY2lrZ2U4MWI4MDA4bHVjajc1am1weTM2aSJ9.JHiBmawEKGsn3jiRK_d0Gw'
```
Comment (JavierJia) [inline src/main/resources/twitter-demo/static/js/newTweetBook.js]: is there any ways to hide the `id` and `accessToken`?

File: src/main/resources/twitter-demo/static/js/newTweetBook.js [hunk 7]
```diff
@@ -0,0 +1,655 @@
...
+  // const
+  colors = [ '#053061','#2166ac','#4393c3','#92c5de','#d1e5f0' ,'#f7f7f7' ,'#fddbc7','#f4a582','#d6604d', '#b2182b','#67001f' ];
+  start_date = "2015-12-17T00:00:00.000Z";
+  end_date = "2016-02-13T23:59:59.000Z";
+
+  // data
+  timeSeries_date = null
+
+  // style
+    stateStyle = {
```
Comment (JavierJia) [inline src/main/resources/twitter-demo/static/js/newTweetBook.js]: should this two styles (`stateStyle`, `countyStyle`) goes to css file?

File: src/main/resources/twitter-demo/static/js/newTweetBook.js [hunk 8]
```diff
@@ -0,0 +1,655 @@
...
+        opacity: 1,
+        color: '#92c5de',
+        fillOpacity: 0.2
+    }
+
+  // Interaction function
+  function highlightFeature(e) {
+    var layer = e.target;
+
+    layer.setStyle({
```
Comment (JavierJia) [inline src/main/resources/twitter-demo/static/js/newTweetBook.js]: give this style a name?

**Comment by JavierJia** [comment]
LGTM. 
General comments:
1. Use a code formatting tool to unify the format.
2. Replace the `tweetbook` tmp + js instead of creating a new one.

File: src/main/resources/twitter-demo/newTweetBook.tpl (hunk 1 above)
Comment (JeremyLi28) [inline src/main/resources/twitter-demo/newTweetBook.tpl]: I will give it a title

File: src/main/resources/twitter-demo/newTweetBook.tpl (hunk 3 above)
Comment (JeremyLi28) [inline src/main/resources/twitter-demo/newTweetBook.tpl]: line 9 is the css file, here is the js file

File: src/main/resources/twitter-demo/newTweetBook.tpl (hunk 4 above)
Comment (JeremyLi28) [inline src/main/resources/twitter-demo/newTweetBook.tpl]: mapbox-gl add webGL to mapbox, but actually we will use neither of them, I will remove them.

File: src/main/resources/twitter-demo/run_tweetbook_demo.py (hunk 5 above)
Comment (JeremyLi28) [inline src/main/resources/twitter-demo/run_tweetbook_demo.py]: Got it

File: src/main/resources/twitter-demo/static/js/newTweetBook.js (hunk 6 above)
Comment (JeremyLi28) [inline src/main/resources/twitter-demo/static/js/newTweetBook.js]: I will look into that

File: src/main/resources/twitter-demo/static/js/newTweetBook.js (hunk 8 above)
Comment (JeremyLi28) [inline src/main/resources/twitter-demo/static/js/newTweetBook.js]: Got it

File: src/main/resources/twitter-demo/static/js/newTweetBook.js (hunk 6 above)
Comment (JeremyLi28) [inline src/main/resources/twitter-demo/static/js/newTweetBook.js]: Seems that public token is safe.
https://www.mapbox.com/help/define-access-token/

**Pull request closed without merging**